engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -288,23 +292,33 @@ class Compiler(
     if (irCachingEnabled) {
       requiredModules.foreach { module =>
         if (!module.hasCrossModuleLinks) {
-          val flags =
-            module.getIr.preorder.map(_.passData.restoreFromSerialization(this))
-
-          if (!flags.contains(false)) {
+          if (module.getIr == null) {
```

Comment:
Shouldn't this condition trigger parsing? If IR is `null` it will likely crash soon anyway?

Reply:
OK, I see we have a couple of places where this condition is checked.